hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/TypeMapping.md

```diff
@@ -0,0 +1,49 @@
+Type mapping in MongoDB SQL Connector.
+=====
+
+The type system in MongoDB and SQL is not exactly the same. That
+leads to potential confusions and the need of type coercion. 
+
+Type mapping
+----
+
+| BSON Type           | SQL Type                | Java type                      |
+|---------------------|-------------------------|--------------------------------|
+| double              | double                  | double                         |
+| string              | varchar                 | String                         |
+| object              | object                  | org.bson.Document              |
+| array               | object                  | List                           |
+| binData             | -                       | -                              |
+| undefined           | -                       | -                              |
+| ObjectId            | object                  | org.bson.ObjectId              |
+| bool                | boolean                 | boolean                        |
+| date                | date_time* or timestamp | LocalDateTime*                 |
+| timestamp           | date_time or timestamp  | LocalDateTime                  |
+| null                |                         |                                |
+| regex               | OBJECT                  | org.bson.BsonRegularExpression |
+| dbPointer           | -                       | -                              |
+| javascript          | varchar                 | String                         |
+| javascriptWithScope | object                  | org.bson.CodeWithScope         |
+| symbol              | -                       | -                              |
+| int (32 bit)        | int                     | int                            |
+| long (64 bit)       | bigint                  | long                           |
+| decimal (128 bit)   | decimal                 | BigDecimal                     |
+| minKey              | object                  | org.bson.MinKey                |
+| maxKey              | object                  | org.bson.MaxKey                |
+
+
+`*` - Type `date` in BSON represents seconds from Unix epoch in UTC timezone. 
+Therefore it's not mapped to pure `DATE` SQL type nor `LocalDate` 
+in Java (nor any formats with timezones).
+
+The "Java Type" column represents an object returned by SQL query
+if the object put into the collection is of given BSON type.
+
+Bare in mind, that, while we are able to convert MongoDB type to requested SQL type
+in the projection, the argument binding will not always work the same 
+due to technical limitations. 
+
+For example: you can have object with type `timestamp` represented as `DATE_TIME`, 
+that in SELECT will give you LocalDateTime. However, binding LocalDateTime
+as an argument won't work, as only native MongoDB types will work for arguments.
+Same for e.g. having BSON column of type "string" mapped to "integer" in SQL.
```

Comment:
As in the table above, ObjectId is mapped to Object type and you can bind it (see AllTypesInsertMongoSqlConnectorTest).

For string/int the reason of fail will be the fact that `select` can convert "1" string to 1 number, but if you bind "1" as parameter where int is required, we won't be able to perform coercion. ExpressionEvaluationContext will just return string - as user passed.

However, then I think about it, we can get the column types, so that may be fixed. I'll try to do it, but I'd rather want to do it in separate PR to reduce scope of this one, WDYT?